scripts/e2e-test.sh

```diff
@@ -142,11 +146,32 @@ fi
 
 export e2e_docker_registry="$registry" # can be empty string
 
-if [ -z "$tests" ]; then
-  tests="$TESTS"
-  if [ -n "$run_extended_tests" ]; then
-    tests="$tests $EXTENDED_TESTS"
+if [ -n "$profile" ]; then
```

Comment:
I think for clarity we should 
1) check and exit if  `$tests` and `$profile` are both set.
2) Default the value of `$profile` to `ondemand`, if `$profile` is empty.

Then it is eminently clear which profile - if any - is being used,
in later parts of the script (if required)

Reply:
Maybe the default profile should be "default" - "ondemand" shouldn't be special.
"--profile default" is the same as --profile being absent, in which case $profile is set to "default"

Reply:
OTOH allowing "--profile default" is maybe going too far.

Reply:
1. Done,
2. Now has a profile named "default" if profile is not specified.